crates/loro-internal/src/state/tree_state.rs

```diff
@@ -41,6 +41,10 @@ pub struct TreeState {
     idx: ContainerIdx,
     trees: FxHashMap<TreeID, TreeStateNode>,
     children: TreeChildrenCache,
+    /// Whether the tree has fractional index.
+    ///
+    /// If false, the fractional index is always [`FractionalIndex::place_holder`]
+    with_fractional_index: bool,
```

Comment:
Should it be Arc<AtomicBool>? It's probably better if it shares this state with configure